dev/src/timestamp.ts

```diff
@@ -107,7 +107,9 @@ export class Timestamp implements firestore.Timestamp {
    */
   static fromMillis(milliseconds: number): Timestamp {
     const seconds = Math.floor(milliseconds / 1000);
-    const nanos = (milliseconds - seconds * 1000) * MS_TO_NANOS;
+    const nanos = Math.round(
```

Comment:
I think you need to round down here so that nanos is always less than "one second in nanos".